File: exercises/beecrowd/1478/1478.go
// https://judge.beecrowd.com/pt/problems/view/1478
package main

import (
	"fmt"
	"math"
)

func main() {
	var n int
	for {
		fmt.Scan(&n)
		if n == 0 {
			break
		}

		matrix := make([][]int, n)
		for i := 0; i < n; i++ {
			matrix[i] = make([]int, n)
			for j := 0; j < n; j++ {
				matrix[i][j] = int(math.Abs(float64(i-j))) + 1
			}
		}

		for i := 0; i < n; i++ {
			for j := 0; j < n; j++ {
				if j > 0 {
					fmt.Printf(" ")
				}
				fmt.Printf("%3d", matrix[i][j])
			}
			fmt.Println()
		}
		fmt.Println()
	}
}

File: exercises/beecrowd/1478/1478.ts
// https://judge.beecrowd.com/pt/problems/view/1478
import * as fs from "fs";

function main(): void {
  const input = fs
    .readFileSync("/dev/stdin", "utf8")
    .trim()
    .split("\n")
    .map(Number);
  let lineIndex = 0;

  while (true) {
    const n = input[lineIndex++];
    if (n === 0) {
      break;
    }

    const matrix: number[][] = [];
    for (let i = 0; i < n; i++) {
      matrix[i] = [];
      for (let j = 0; j < n; j++) {
        const value = Math.abs(i - j) + 1;
        matrix[i][j] = value;
      }
    }

    for (let i = 0; i < n; i++) {
      let rowString = "";
      for (let j = 0; j < n; j++) {
        rowString += String(matrix[i][j]).padStart(3, " ");
        if (j < n - 1) {
          rowString += " ";
        }
      }
      console.log(rowString);
    }
    console.log();
  }
}

main();
